### ISSUE TITLE
Incorrect data is retrieved in Resultset.

### ISSUE THREAD

**Issue by podoni** [dbeaver/dbeaver#19848]
### Description

A character value that cannot be expressed as a result value is not null was retrieved.
When I save it as Excel, "????" expressed in text.
When I checked with the nvl function, it wasn't even null data.
It is expressed as ㅁㅁㅁㅁㅁ in the value panel. The actual data is string data in the form of a date. (ex:20230301)

While writing sql, the data that should come out was not visible, so I found it through various tests.
Actually, all I modified was to remove the "1 = 1" statement from "on 1 = 1".
After editing, the data I wanted was retrieved.

### [before editing]
![image](https://user-images.githubusercontent.com/59002295/235590996-25792791-8bdd-4f43-bc5b-9bdec23e3736.png)

### [after editing]
![image](https://user-images.githubusercontent.com/59002295/235591022-55ff620f-88e5-420f-bd24-c97186281f6f.png)

### [Save Excel]
![image](https://user-images.githubusercontent.com/59002295/235590620-0ab85865-aaf1-4d1b-b17e-7817902ab79b.png)

Thank for Reading. :)

### DBeaver Version

Community Edition 23.0.3

### Operating System

Windows 10

**Comment by ShadelessFox** [comment]
Hello @podoni,

Can you please provide more information regarding your database and the driver used for connecting?

**Comment by podoni** [comment]
Hello @ShadelessFox 
Thank for reply.

**Server : Oracle Oracle Database 18c Enterprise Edition Release 18.0.0.0.0 - Production Version 18.8.0.0.0
Driver : Oracle JDBC driver 18.3.0.0.0**


Please let me know if you need additional information to resolve this issue. 

:)

**Comment by ShadelessFox** [comment]
It looks like temporal data was interpreted as a string instead of a date/time value.

Can you please tell what is the type of the `BASEDD` column before and after the change? Use the metadata panel on the right.

**Comment by podoni** [comment]
@ShadelessFox
When I checked the meta information, it was confirmed as varchar2.
![image](https://github.com/dbeaver/dbeaver/assets/59002295/aa3c5b33-8676-40ce-ab08-97317057110c)

**Comment by ShadelessFox** [comment]
What about the other case where the data is correctly displayed?

**Comment by podoni** [comment]
@ShadelessFox
metadata is expressed identically.

![image](https://github.com/dbeaver/dbeaver/assets/59002295/ca163298-4d8f-4810-8c51-35fb03b3abb1)

**Comment by ShadelessFox** [comment]
1. What type is shown for the `BASEDD` column in the table's DDL?
2. Could you please send the log files? You can find instructions here: https://dbeaver.com/docs/wiki/Log-files/

**Comment by podoni** [comment]
@ShadelessFox

I looked at the log file (.log) and there was no additional content.
I've executed many SQL in the meantime, but this was the first time.
I talked about the problem situation, but it is difficult to reproduce, so I hope the issue ends at this point.
I understand this situation and will use it with care in the future.
Thank you for caring about me all this time.

I am always grateful for making such a great program available.
